src/buildstream/sandbox/_sandboxremote.py

```diff
@@ -294,12 +294,20 @@ def _execute_action(self, action, flags):
 
         # Fetch outputs
         for output_directory in action_result.output_directories:
+            # Now do a pull to ensure we have the full directory structure.
+            # We first try the root_directory_digest we requested, then fall back to tree_digest
+
+            root_directory_digest = output_directory.root_directory_digest
+            if root_directory_digest and root_directory_digest.hash:
+                cascache._fetch_directory(casremote, root_directory_digest)
```

Comment:
Nit, `_fetch_directory()` should be marked as public (internal) by removing the leading underscore as we're calling it from outside `cascache.py`.